File: Angular/cdb-project/src/app/model/menu-model.ts
export class MenuModel{
    menuId!: number;
    menuName: string = '';
    parentMenuId!: number;
	programCode: string = '';
	orderNo: string = '';
	level!: number;
	iconName: string = '';
	activeFlag: string = '';
    menus: Array<MenuModel> = [];
}

File: Spring/Model/MenuModel.java
package pccth.sp.pccthspseedservice.model;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class MenuModel {
	Integer menuId;
	String menuName;
	Integer parentMenuId;
	String programCode;
	String orderNo;
	Integer level;
	String iconName;
	String activeFlag;
	
	List<MenuModel> menus;
}
